tests/src/rules/no-self-import.js

```diff
@@ -0,0 +1,42 @@
+import { test, testFilePath } from '../utils'
+
+import { RuleTester } from 'eslint'
+
+const ruleTester = new RuleTester()
+    , rule = require('rules/no-self-import')
+
+const error = {
+  ruleId: 'no-self-import',
+  message: 'Module imports itself',
+}
+
+ruleTester.run('no-self-import', rule, {
+  valid: [
+    test({ code: 'import _ from "lodash"'}),
```

Comment:
how can any of these valid examples be deemed such without a filename to test against?

It's probably good to keep all of these, to ensure that when there's no filename (like stdin input), the rule bails out, but these should include a filename explicitly.